pennylane/gradients/finite_difference.py

```diff
@@ -0,0 +1,223 @@
+    b = np.zeros_like(shifts)
+    b[n] = factorial(n)
+    coeffs = np.linalg.solve(A, b)
+
+    stencil = np.stack([coeffs, shifts])
+    stencil[0, np.abs(stencil[0, :]) < 1e-10] = 0
+    stencil = stencil[:, ~np.all(stencil == 0, axis=0)]
+    stencil = stencil[:, np.argsort(np.abs(stencil)[1])]
+    return stencil
+
+
+def generate_shifted_tapes(tape, idx, shifts):
+    r"""Generate a list of tapes where the corresponding trainable parameter
+    index has been shifted by the values given.
+
+    Args:
+        tape (.QuantumTape): input quantum tape
+        idx (int): trainable parameter index to shift the parameter of
+        shifts (Sequence[float or int]): sequence of shift values
+
+    Returns:
+        list[QuantumTape]: List of quantum tapes. Each tape has parameter
+        ``idx`` shifted by consecutive values of ``shift``. The length
+        of the returned list of tapes will match the length of ``shifts``.
+    """
+    params = qml.math.stack(tape.get_parameters())
+    tapes = []
+
+    for s in shifts:
+        shifted_tape = tape.copy(copy_operations=True)
+
+        shift = np.zeros(qml.math.shape(params), dtype=np.float64)
+        shift[idx] = s
+
+        shifted_params = params + qml.math.convert_like(shift, params)
+        shifted_tape.set_parameters(qml.math.unstack(shifted_params))
+
+        tapes.append(shifted_tape)
+
+    return tapes
+
+
+def finite_diff(tape, argnum=None, h=1e-7, order=1, n=1, form="forward"):
+    r"""Generate the parameter-shift tapes and postprocessing methods required
+    to compute the gradient of an gate parameter with respect to an
+    expectation value.
+
+    Args:
+        tape (.QuantumTape): quantum tape to differentiate
+        argnum (int or list[int] or None): Trainable parameter indices to differentiate
+            with respect to. If not provided, the derivative with respect to all
+            trainable indices are returned.
+        h (float): finite difference method step size
+        order (int): The order of the finite difference method to use.
+        n (int): compute the :math:`n`-th derivative
+        form (str): The form of the finite difference method. Must be one of
+            ``"forward"``, ``"center"``, or ``"backward"``.
+
+    Returns:
+        tuple[list[QuantumTape], function]: A tuple containing a
+        list of generated tapes, in addition to a post-processing
+        function to be applied to the evaluated tapes.
+
+    **Example**
+
+    >>> with qml.tape.QuantumTape() as tape:
+    ...     qml.RX(params[0], wires=0)
+    ...     qml.RY(params[1], wires=0)
+    ...     qml.RX(params[2], wires=0)
+    ...     qml.expval(qml.PauliZ(0))
+    ...     qml.var(qml.PauliZ(0))
+    >>> tape.trainable_params = {0, 1, 2}
+    >>> gradient_tapes, fn = qml.gradients.finite_diff.grad(tape)
+    >>> res = dev.batch_execute(gradient_tapes)
+    >>> fn(res)
+    [[-0.38751721 -0.18884787 -0.38355704]
+     [ 0.69916862  0.34072424  0.69202359]]

```

Comment:
Re. my comment in the changelog, might be helpful to add those extra details regarding the form of the output here.

Reply:
Have added it in!